File: JUNGOL/Beginner/#5947 숫자 삼각형3/숫자 삼각형3.cpp
#include <iostream>

using namespace std;

int main() {
	int n; cin>>n;

	if(n<1 || n>50 || !(n%2)) cout<<"INPUT ERROR!\n";
	else{
		for(int i=0; i<n/2+1; i++){
			for(int j=0; j<=i; j++) printf("%d ", j+1);
			cout<<endl;
		}

		for(int i=n/2; i>=1; i--){
			for(int j=0; j<i; j++) printf("%d ", j+1);
			cout<<endl;
		}
	}
}

File: JUNGOL/Beginner/#5947 숫자 삼각형3/숫자 삼각형3.py
def main():
    n = int(input())

    if n<1 or n>50 or n%2==0: print("INPUT ERROR!")
    else:
        for i in range(n//2+1):
            for j in range(i+1):
                print(j+1, end=' ')
            print()

        for i in range(n//2, -1, -1):
            for j in range(i):
                print(j+1, end=' ')
            print()

main()
